src/jinja2/parser.py

```diff
@@ -266,6 +267,18 @@ def parse_block(self):
             )
 
         node.body = self.parse_statements(("name:endblock",), drop_needle=True)
+
+        # enforce that required blocks only contain whitespace or comments
+        # by asserting that the body, if not empty, is just TemplateData nodes
+        # with whitespace data
+        if node.required:
+            try:
+                assert all(
```

Comment:
Don't use an assert here, it will be optimized out when running `python -O`. flake8 is supposed to catch this.

Reply:
👍 fixed. (accidentally resolved without actually pushing the changes before)